msteams-platform/samples/virtual-assistant.md

```diff
@@ -0,0 +1,560 @@
+
+## Implementation considerations
+
+The decision to add a Virtual Assistant can include many determinants and differ for each organization. Here are the factors that support implementing a Virtual Assistant for your organization:
+
+- A central team manages all employee experiences and has the capability to build a Virtual Assistant experience and manage updates to the core experience including the addition of new skills.
+- Multiple applications exist across business functions and/or the number is expected to grow in the future.
+- Existing applications are customizable, owned by the organization, and can be converted into skills for a Virtual Assistant.
+- The central employee-experiences team is able to influence customizations to existing apps and provide necessary guidance for integrating existing applications as skills in Virtual Assistant experience
+
+![Central team maintains the assistant, and business function teams contribute skills](../assets/images/bots/virtual-assistant/business-functions.png)
+
+## Create a Teams-focused Virtual Assistant
+
+Microsoft has published a [Visual Studio template](https://marketplace.visualstudio.com/items?itemName=BotBuilder.VirtualAssistantTemplate) for building Virtual Assistants and skills. With the Visual Studio template, you can create a Virtual Assistant, powered by a text-based experience with support for limited rich cards with actions. We have enhanced the Visual Studio base template to include Microsoft Teams platform capabilities and power great Teams app experiences. A few of the capabilities include support for rich adaptive cards, task modules, teams/group chats and messaging extensions. *See also*, [Tutorial: Extend Your Virtual Assistant to Microsoft Teams](https://microsoft.github.io/botframework-solutions/clients-and-channels/tutorials/enable-teams/1-intro/).
+
+![High-level diagram of a Virtual Assistant solution](../assets/images/bots/virtual-assistant/high-level-diagram.png)
+
+### Add adaptive cards to your Virtual Assistant
+
+To dispatch requests properly, your Virtual Assistant needs to identify the correct LUIS model and corresponding skill associated with it. However, the dispatching mechanism cannot be used for card action activities since the LUIS model associated with a skill may not be trained for card action texts since these are fixed, pre-defined keywords, not utterances from a user.
+
+We have resolved this by embedding skill information in the card action payload. Every skill should embed `skillId` in the  `value` field of card actions. This is the best way to ensure that each card action activity carries the relevant skill information and Virtual Assistant can utilize this information for dispatching.
+
+Below is a card action data sample. **By providing `skillId` in the constructor we ensure that skill information is always present in card actions.**
```

Comment:
@laujan @nebhagat Is the bolding here (and in a few phrases in the subsequent paragraphs) intentional?

Reply:
Yes, not needed, hence removed from here and from subsequent paragraphs.